tab-oneipl/minutes/2022_02_17_Minutes.rst

```diff
@@ -0,0 +1,183 @@
+  the get_pointer() is replaced by general get_access(args) function, which is
+  common for USM and Image (Texture) access abstraction.
+
+  "unspecified" return type of this API is not language-related, but a
+  spec-related placeholder for the return type, as per SYCL specification.
+  It can be defined differently for particular class specialization in the
+  implementation.
+
+  The minimal API approach is applied for Image class per oneIPL TAB feedbacks:
+  get_rect(), get_roi() and get_origin() functions are left to get access to
+  the necessary part of image, and the size accessors are now moved to the
+  roi_rect struct (see slide 7 of `oneIPL TAB #3`_ for the details).
+
+* **Mark Rabotnikov, Philips**: Why do we need get_origin()? Is not it the same
+  as image itself?
+
+  **Victor Getmanskiy, Intel**: yes, it may be needed in user kernels to use
+  neighbor pixels to process borders for example.
+  It is common case for Filters, when you need to access the pixels outside the
+  processing area. Also, it may be used to create a different ROI on the same
+  origin.
+
+
+oneIPL Memory Model:
+
+* oneIPL spec supports different type of memory assigned via AllocatorT or
+  provided via pointer by user to image constructor: host USM, device USM,
+  shared USM and Image (texture) memory.
+
+  USM allocator has a major difference from std::allocator. Allocator cannot be
+  constructed without context, so device context is required. It complicates
+  the APIs, since allocator has no default constructor.
+
+* Victor Getmanskiy presented the different ways of memory allocation on host
+  and device and different types of access and control (see slide 9 for
+  details)
+
+* **Ashish Uthama, Mathworks**: What happens at the point of access?
+  Is the copy operation performed?
+
+  **Victor Getmanskiy, Intel**: It depends on the allocation and access types:
+   * If you have device allocation and memory assess on device -> no copy is
+     performed.
+   * If you have shared allocator and access for read -> no copy as well, but
+     for the write access copy to host would happen after kernel is finished.
+   * In image_allocator_t case the memory will be allocated on both host
+     and device and access is possible only to write or to read, but not both.
+     In case of read no copy to host happen, in case of write copy to host
+     would happen after kernel is finished.
+
+* **Ashish Uthama, Mathworks**: Actually is this SYCL limitation? It is not so
+  clear: if you have reading only, you do not know if it do the copy or not.
+  I prefer the explicit copy operation, so I am wondering if this is
+  restriction of SYCL?
+
+  **Victor Getmanskiy, Intel**: For the USM it is possible to implement
+  explicit copy, for Image it is under question. You still can explicitly copy
+  USM memory and create image with device_allocator_t over memory without copy.
+  For Textures it is currently not possible, because of SYCL limitations, but
```

Comment:
```suggestion
  For Images(Textures) creating object over device memory is currently not possible, only construction from host pointer is available, because of SYCL limitations. But
```